extension/src/entrypoints/append-test-case.ts

```diff
@@ -0,0 +1,188 @@
+// extension/src/entrypoints/append-test-case.ts
+import { ExtractMessage, MessagePayload, WindowMessage } from "@cb/types";
+import {
+  waitForElementAtIndex,
+  waitForElementsWithCondition,
+} from "@cb/utils/dom";
+
+const appendTestCaseToLeetCode = async (
+  args: MessagePayload<
+    ExtractMessage<WindowMessage, "appendTestCaseToLeetCode">
+  >
+) => {
+  const { testValues } = args;
+
+  try {
+    const testCaseButtons = document.querySelectorAll(
+      'button[data-e2e-locator="console-testcase-tag"]'
+    );
+
+    if (testCaseButtons.length === 0) {
+      throw new Error("No test case buttons found");
+    }
+
+    const lastTestCaseButton = testCaseButtons[
+      testCaseButtons.length - 1
+    ] as HTMLElement;
+    const container = lastTestCaseButton.closest("div")?.parentElement;
+
+    let addButton: HTMLButtonElement | undefined;
+
+    if (container) {
+      const closedButtons = container.querySelectorAll(
+        'button[data-state="closed"]'
+      );
+      addButton = Array.from(closedButtons)[0] as HTMLButtonElement | undefined;
+    }
```

Comment:
Can just collapsed the entire thing

```suggestion
const button = lastTestCaseButton.parentElement?.querySelector('button[data-state="closed"]')

if (button == null) return;
```